hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioOutboundPipeline.java

```diff
@@ -155,26 +147,18 @@ public void write(OutboundFrame frame) {
     }
 
     private OutboundFrame poll() {
-        for (; ; ) {
-            OutboundFrame frame = urgentWriteQueue.poll();
+        OutboundFrame frame = urgentWriteQueue.poll();
+        if (frame == null) {
+            frame = writeQueue.poll();
             if (frame == null) {
-                frame = writeQueue.poll();
-                if (frame == null) {
-                    return null;
-                }
-                normalFramesWritten.inc();
-            } else {
-                priorityFramesWritten.inc();
+                return null;
             }
-
-            if (frame.getClass() == TaskFrame.class) {
```

Comment:
All tasks are now executed using the NioPipelineTask; which is part of the super class. We don't need to deal with tasks hidden as packets any longer.